tests/unit_tests/core/tuple_accessor.cpp

```diff
@@ -0,0 +1,41 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <iostream>
+
+#include "detray/definitions/detail/accessor.hpp"
+
+// Thrust include(s).
+#include <thrust/tuple.h>
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+TEST(tuple_accessor, tuple_accessor) {
+
+    using namespace detray;
+
+    // std::tuple test
+    auto s_tuple = detail::make_tuple<std::tuple>(1.0, 2, "std::tuple");
+
+    const int s_tuple_size = detail::tuple_size<decltype(s_tuple)>::value;
+
+    EXPECT_EQ(s_tuple_size, 3);
+    EXPECT_EQ(detail::get<0>(s_tuple), 1.0);
+    EXPECT_EQ(detail::get<1>(s_tuple), 2);
+    EXPECT_EQ(detail::get<2>(s_tuple), "std::tuple");
+
+    // thrust::tuple test
+    auto t_tuple = detail::make_tuple<thrust::tuple>(1.0, 2, "thrust::tuple");
+
+    const int t_tuple_size = detail::tuple_size<decltype(t_tuple)>::value;
```

Comment:
This could be turned into `auto` as well.

I do know that once I'll try to build this code on Windows with MSVC (Oh, be scared. It **is** coming. :stuck_out_tongue:), that will freak out about such implicit type conversions.

Reply:
Yeah, that might be better